language/move-prover/bytecode-to-boogie/test_mvir/verify-synthetics.mvir

```diff
@@ -0,0 +1,60 @@
+// Tests basic invariant functionality.
+
+module TestInvariants {
+
+    synthetic syn: u64;
+
+    // Its not yet clear what's the start value of synthetics is, so we use this to initialize it.
+    struct CrudeWayToInitializeSyn {
```

Comment:
Why does it have to be inside a struct?

Reply:
I just pushed an update which eliminates the need for this. We can actually use an arbitrary start value for synthetics and write: 
```
public valid()
    ensures syn == old(syn) + 2 + 3 - 3 + 4 - 2
{ ... }
```

If you mean why do we write invariants inside of structs: we don't have to but they are still associated with types. So if we write them outside, we would have to say to which type they apply.

Reply:
Cool. I was just asking why we were initializing syn inside a struct. I understood why invariants associated with types are inside the structs.